core/test/lib/libledger-test/ProxyHttpClient.cpp

```diff
@@ -1,28 +1,254 @@
-#include "ProxyHttpClient.hpp"
+﻿#include "ProxyHttpClient.hpp"
+#include "api/Error.hpp"
+#include "api/HttpMethod.hpp"
 #include "api/HttpRequest.hpp"
+#include <api/HttpReadBodyResult.hpp>
+#include <experimental/filesystem>
+#include <algorithm>
+#include <fstream>
+#include <functional>
 #include <memory>
-#include "api/Error.hpp"
+#include <mutex>
+#include <gtest/gtest.h>
+
+#define HTTP_CACHE_DIR "http_cache"
+
+#if defined(UPDATE_HTTP_CACHE) && defined(ALLOW_HTTP_ACCESS) && !defined(LOAD_HTTP_CACHE)
+#pragma message "ProxyHttpClient is configured to update HTTP cache"
+#elif !UPDATE_HTTP_CACHE && ALLOW_HTTP_ACCESS && !LOAD_HTTP_CACHE
+#pragma message "ProxyHttpClient is configured to run tests agains real web services"
+#elif !UPDATE_HTTP_CACHE && !ALLOW_HTTP_ACCESS && LOAD_HTTP_CACHE
+#pragma message "ProxyHttpClient is configured to use HTTP cache instead of real web services"
+#else
+#error "Incorrect configuration of HTTP caching"
+#endif
+
+using namespace std;

```

Comment:
Let's avoid `using namespace std;`, what do you think ?

Reply:
done